crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -993,7 +1011,14 @@ pub fn resolve_assignment<'a>(
             semantic.resolve_qualified_name(func)
         }
         _ => None,
+    }?;
+    for member in reversed_tail.iter().rev() {
+        // extend the qualified name with the attributes we accessed
+        // e.g. for `app.get` when `app` resolves to `fastapi.FastAPI`
+        // then we want to return `fastapi.FastAPI.get`.
+        qualified_name = qualified_name.append_member(member);
     }
```

Comment:
Nit: We could consider adding an `extend` method` to reduce the code

Reply:
I struggled a bit to get the lifetimes correct. If there's a more idiomatic way to write the function, feel free to go ahead and fix it. But I definitely like how it simplified the `resolve_assignment`, I also noticed that the two match arms were redundant, so I consolidated them.

Reply:
The trick was to use `into_iterator` so that the items have the type `&'a str` and not `&&'a str`. The alternative would have been to use `iter().copied()`, I think, which dereferences `&&'a str` to `&'a str`